### PR TITLE
PA-1500 : Add bkp status check in `CreateRestore()`

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
While running tests it was seen that in some instances like show below: 
```
09:13:54  ------------------------------
09:13:54  • Failure [8153.030 seconds]
09:13:54  {PXBackupEndToEndBackupAndRestoreWithUpgrade}
09:13:54  /go/src/github.com/portworx/torpedo/tests/backup/backup_upgrade_test.go:219
09:13:54    PX-Backup End-to-End Backup and Restore with Upgrade [It]
09:13:54    /go/src/github.com/portworx/torpedo/tests/backup/backup_upgrade_test.go:294
09:13:54  
09:13:54    Unexpected error:
09:13:54        <*errors.errorString | 0xc00110e6e0>: {
09:13:54            s: "error: rpc error: code = InvalidArgument desc = backup object [postgres-backup-src-pxbackuptask-0-08-08-01h27m46s-single-namespace-schedule-1691458483-93fd8ce-interval-2023-08-08-031953] status is not in success, Description: Verifying restoration [test-restore-single-ns-iwhp] of next single namespace schedule backup [postgres-backup-src-pxbackuptask-0-08-08-01h27m46s-single-namespace-schedule-1691458483-93fd8ce-interval-2023-08-08-031953] in cluster [test-restore-single-ns-iwhp]",
09:13:54        }
09:13:54        error: rpc error: code = InvalidArgument desc = backup object [postgres-backup-src-pxbackuptask-0-08-08-01h27m46s-single-namespace-schedule-1691458483-93fd8ce-interval-2023-08-08-031953] status is not in success, Description: Verifying restoration [test-restore-single-ns-iwhp] of next single namespace schedule backup [postgres-backup-src-pxbackuptask-0-08-08-01h27m46s-single-namespace-schedule-1691458483-93fd8ce-interval-2023-08-08-031953] in cluster [test-restore-single-ns-iwhp]
09:13:54    occurred
09:13:54  
09:13:54    /go/src/github.com/portworx/torpedo/pkg/aetosutil/dashboardutil.go:427
09:13:54  
09:13:54    Full Stack Trace\
09:13:54    github.com/portworx/torpedo/pkg/aetosutil.(*Dashboard).VerifyFatal(0x73a3e1a?, {0x67d17e0?, 0xc002050398}, {0x0?, 0x0}, {0xc00158a900, 0x102})
09:13:54    	/go/src/github.com/portworx/torpedo/pkg/aetosutil/dashboardutil.go:427 +0x143
09:13:54    github.com/portworx/torpedo/tests/backup.glob..func74.2.15.1(0x0, {0xc001678a80, 0x32})
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_upgrade_test.go:649 +0x7a5
09:13:54    reflect.Value.call({0x636ef20?, 0xc000f40b80?, 0x7f5c2c502108?}, {0x7242eae, 0x4}, {0xc000fafd70, 0x2, 0x0?})
09:13:54    	/usr/local/go/src/reflect/value.go:584 +0x8c5
09:13:54    reflect.Value.Call({0x636ef20?, 0xc000f40b80?, 0x1?}, {0xc000fafd70?, 0x20?, 0x7f5c2c502108?})
09:13:54    	/usr/local/go/src/reflect/value.go:368 +0xbc
09:13:54    github.com/portworx/torpedo/tests/backup.TaskHandler.func1({0x6203f60?, 0xa4925a0?, 0x0?}, {0x6209960?, 0xc000edc990?, 0x203000?})
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_helper.go:3139 +0x1a7
09:13:54    github.com/portworx/torpedo/tests/backup.TaskHandler({0x612ada0?, 0xc002002b58?}, {0x636ef20?, 0xc000f40b80}, 0x1)
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_helper.go:3146 +0x83d
09:13:54    github.com/portworx/torpedo/tests/backup.glob..func74.2.15()
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_upgrade_test.go:653 +0x168
09:13:54    github.com/portworx/torpedo/tests/backup.glob..func74.2()
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_upgrade_test.go:609 0x1097
09:13:54    github.com/portworx/torpedo/tests/backup.TestBasic(0x0?)
09:13:54    	/go/src/github.com/portworx/torpedo/tests/backup/backup_basic_test.go:64 +0xb9
09:13:54    testing.tRunner(0xc000df04e0, 0x7462ca0)
09:13:54    	/usr/local/go/src/testing/testing.go:1446 +0x10b
09:13:54    created by testing.(*T).Run
09:13:54    	/usr/local/go/src/testing/testing.go:1493 +0x35f
09:13:54  ------------------------------
```

Changes done:
- [x] Adding additional code to check backup status before we attempt a restore in `CreateRestore()`

Dashboard URL : 
BasicBackupCreation with s3 bucket: https://aetos.pwx.purestorage.com/resultSet/testSetID/306420 
BasicBackupCreation with nfs: https://aetos.pwx.purestorage.com/resultSet/testSetID/306467 
Full Run with s3 : https://aetos.pwx.purestorage.com/resultSet/testSetID/306822 

**Which issue(s) this PR fixes** (optional)
Closes # PA-1500

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -763,6 +763,26 @@ func CreateRestore(restoreName string, backupName string, namespaceMapping map[s
...
+		Uid:   bkpUid,
+		OrgId: orgID,
+	}
+	resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return err
+	}
+	actual := resp.GetBackup().GetStatus().Status
+	reason := resp.GetBackup().GetStatus().Reason
+	if actual == api.BackupInfo_StatusInfo_Success {
```
Comment (snigdha-px) [inline tests/backup/backup_helper.go:783]: Should it not be "!="?

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -763,6 +763,26 @@ func CreateRestore(restoreName string, backupName string, namespaceMapping map[s
...
+		return fmt.Errorf("backup status for [%s] expected was [%s] but got [%s] because of [%s]", backupName, api.BackupInfo_StatusInfo_Success, actual, reason)
+	}
 	backupDriver := Inst().Backup
 	log.Infof("Getting the UID of the backup %s needed to be restored", backupName)
```
Comment (snigdha-px) [inline tests/backup/backup_helper.go:787]: Also , we can remove this as we are getting the UID  in previous step now

**Review by snigdha-px — CHANGES_REQUESTED** [review]

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go:783]: Done

File: tests/backup/backup_helper.go (hunk 2 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go:787]: Done

**Review by snigdha-px — APPROVED** [review]
LGTM

**Comment by kphalgun-px** [comment]
1. You could have utilized the backupSuccessCheck function.
2. When we try to restore a failed backup, the test case will fail due to either this check or PxBackup throwing an error. Given this, should we proceed with adding the check?

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Comment by kshithijiyer-px** [comment]
1. You could have utilized the backupSuccessCheck function. -> Check runs with DoRetryWithTimeout it'll be an unnecessary loop which will only run once and bail out. If the backup is still in progress this will not bail but keep waiting and then it'll attempt a restore. Instead of this what we want here is to bail out immediately if the backup is any other state in any other state than success.  

2.  When we try to restore a failed backup, the test case will fail due to either this check or PxBackup throwing an error. Given this, should we proceed with adding the check? ->  You can't really restore a failed backup 

@kphalgun-px let me know if you still feel otherwise.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**